modin/core/io/text/text_file_dispatcher.py

```diff
@@ -267,9 +277,8 @@ def partitioned_file(
 
         file_size = cls.file_size(f)
 
-        rows_skipper(header_size)
-
         if pre_reading:
```

Comment:
can we rewrite this part like that?
```python
pd_df_metadata = None
if pre_reading:
    rows_skipper(header_size)
    ...
    rows_skipper(skiprows)
else:
    rows_skipper(skiprows)
    if read_callback_kw:
        start = f.tell()
        ...
    rows_skipper(header_size)
```
IMO it would be much easier to understand